server/src/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -1746,10 +1747,10 @@ public boolean storagePoolHasEnoughSpace(List<Volume> volumes, StoragePool pool,
                     allocatedSizeWithTemplate = _capacityMgr.getAllocatedPoolCapacity(poolVO, tmpl);
                 }
             }
-
-            if (volumeVO.getState() != Volume.State.Ready) {
-                totalAskingSize += getDataObjectSizeIncludingHypervisorSnapshotReserve(volumeVO, pool);
-
+            // A ready state volume is already allocated in a pool. so the asking size is zero for it.
+            // In case the volume is moving across pools or is not ready yet, the asking size has to be computed
+            s_logger.debug("pool id for the volume with id: " + volumeVO.getId() + " is: " + volumeVO.getPoolId());
```

Comment:
Please wrap this `DEBUG` log in an `if (s_logger.isDebugEnabled)` check to prevent unnecessary/expensive string concatenation when `DEBUG` logging is not enabled.

Minor nit: grammatically, the `:` character after `is` is unnecessary.